### PR TITLE
Re-implementing GeoUtils back in Java

### LINKED ISSUE #253: Reintroduce `GeoUtils` as Java class
Since `GeoUtils.java` is used in PSDM, we need to keep a Java version. So I would argue for turning the Scala class back into a Java class

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #253

### CODE REVIEW

File: src/main/java/edu/ie3/util/geo/GeoUtils.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+  /**
+   * Calculates the total length of a LineString through building the sum of the distances between
+   * all points of LineString using {@link #calcHaversine(double, double, double, double)}
+   *
+   * @param lineString the line string which length shall be calculated
+   * @return The total length of the line string
+   */
+  @Deprecated(since = "2.0", forRemoval = true)
+  public static ComparableQuantity<Length> totalLengthOfLineString(LineString lineString) {
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/util/geo/GeoUtils.java:205]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/util/geo/GeoUtils.java:205]: @sonatype-lift ignore

File: src/main/java/edu/ie3/util/geo/GeoUtils.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+   *
+   * @param c1 start coordinate of the linestring
+   * @param c2 end coordinate of the linestring
+   * @return A safely build line string
+   */
+  public static LineString buildSafeLineStringBetweenCoords(
+      final Coordinate c1, final Coordinate c2) {
+    final Coordinate safeCoord1 = c1.equals(c2) ? buildSafeCoord(c1) : c1;
+    return DEFAULT_GEOMETRY_FACTORY.createLineString(
+        ArrayUtils.addAll(new Coordinate[] {safeCoord1}, c2));
```
Comment (t-ober) [inline src/main/java/edu/ie3/util/geo/GeoUtils.java:101]:
```suggestion
        ArrayUtils.addAll(safeCoord1, c2));
```

shouldn't this suffice?

**Review by t-ober — CHANGES_REQUESTED** [review]
Thanks a lot for taking this over 😍   
Looks pretty solid

File: src/main/java/edu/ie3/util/geo/GeoUtils.java (hunk 2 above)
Comment (sebastian-peter) [reply src/main/java/edu/ie3/util/geo/GeoUtils.java]: Found an even better solution imho :)

**Review by t-ober — APPROVED** [review]

**Pull request merged**